src/index.js

```diff
@@ -7,12 +7,21 @@ class PromisePool extends Emitter {
    *
    * You should _only_ #flush() after you've have no more functions to start.
    *
-   * @param {Number} concurrentFactor The maximum number of functions to let run concurrently.
+   * @param {Number=} options.numConcurrent The maximum number of functions to let run concurrently.
+   * @param {Number=} options.maxPending The maximum number of pending functions to allow in the
+   *   pending queue, in an attempt to forbid use of promise-pool in incorrect ways.
    */
-  constructor(concurrentFactor) {
+  constructor(options) {
     super();
 
-    this._numConcurrent = asPositiveInteger(concurrentFactor, 4, 1);
+    if (!options) {
```

Comment:
Do you want to support the old style `new PromisePool(10)` as a shorthand? I guess that obfuscates the thing we're trying to make explicit, but it might also simplify the most common use-case.

Reply:
Hmm I was doing this to avoid a major version rev, and because I think this is fine behavior.

Now that I'm thinking about it, though - this _is_ a breaking change and maybe we should do a major rev to avoid breaking code that technically works even if it's incorrect usage.

Reply:
I agree with doing a major change here.

Reply:
Yes, I agree, this is a breaking change (it breaks the "strange usage" case) and semantically should be a major version.

Reply:
7ecb40f